### DISCUSSION TITLE
Automatic HDR tone mapping switching for USB grabber using webos/piccap

### DISCUSSION THREAD

**Discussion by NatyaSadella** [awawa-dev/HyperHDR#532] [Q&A] [answered]
@awawa-dev UPDATE: The same result can be achieved by putting the splitter from copy edid to 4k7.1; with this edid setting HyperHDR automatically switches between HDR on/off in my case. 

I switched over from webOS Piccap to your recommended MS2109 - Ezcoo Splitter Setup.

Everything works great except HDR to SDR switching. When I enable HDR Tonemapping, HDR and DV are 1:1 but SDR looks completely wrong. It seems like something is not telling HyperHDR to turn off HDR.

SDR with Tonemapping enabled:

![shot160002](https://user-images.githubusercontent.com/119701306/225390735-849ccd30-5712-408c-9d79-0668daf8ca02.png)

SDR with Tonemapping disabled:

![shot160051](https://user-images.githubusercontent.com/119701306/225390831-91b4a594-cc44-4d05-9e04-f7c073b77857.png)

What can I do about it?

**Comment by awawa-dev** [comment]
Hi
There is no and never has been automatic SDR/HDR/DV signal detection for the "HDR Tone Mapping" command in HyperHDR. Maybe it will appear in the future. Currently you can control it over the web or use the JSON API for remote control or automate it (e.g. Denon amplifiers are aware of the video format and also expose the API) or force the player/codec to always upconvert SDR to HDR e.g. using madvr.

**Reply by computergreek** [reply]
hey heads up, i think i'm onto something with being able to automate it through home assistant using my TV (Hisense): https://github.com/sehaas/ha_hisense_tv

**Reply by awawa-dev** [reply]
Does it return an SDR or HDR tag in some function in the API?

**Reply by computergreek** [reply]
i haven't had a chance yet to deploy the integration in my home assistant setup--it's on my to-do list, i'm not sure yet when i'll get around to it. it does however look really promising so i thought i'd share in case anybody with a hisense tv wanted to jump on it and test it out

**Comment by awawa-dev** [comment, marked as answer]
If the webos backend correctly detected SDR/HDR and enabled/disabled 'HDR to SDR tone mapping' automatically in HyperHDR, you can still use it even though you're using a USB grabber now. Just set the USB grabber instance to a lower priority than the flatbuffer network video source and make sure piccap connects to HyperHDR instance running on another device equipped with a USB grabber.

**Reply by NatyaSadella** [reply]
Wow thank you I will test this out

**Reply by NatyaSadella** [reply]
So I can't get it to work that way. I set up piccap as 150 and usb grabbing as 100. But flatbuffer doesn't appear in the source menu. Restarted TV and Stopped/Started Piccap. Instead of 127.xxxxx I typed in the IP of HyperHDR USB Grabber. So basically I've tried to have 1 HyperHDR with two sources. But I can't see Piccap Flatbuffer in the source menu.

**Reply by awawa-dev** [reply]
If the flatbuffer source doesn't appear in the remote tab then probably there is no connection from webos. First setup working flatbuffer based config, check if you have a video from piccap in the live preview and then change the flatbuffers priority to higher than USB grabber to switch the source.

**Reply by awawa-dev** [reply]
I know many users use this config: webos backend + HyperHDR on Rpi so you just need to double check everything. Especially piccap config but I cant help with that.

**Reply by NatyaSadella** [reply]
So I've checked the logs and flatbuffer tries to connect but next line is always "connection closed".

**Reply by awawa-dev** [reply]
You need to verify everything yourself or ask for help on some webos channel. This was the base case scenario in the past when you couldn't run HyperHDR directly on the webos and the application had to run on another external unit with flatbuffers as a network video source from piccap.

**Reply by NatyaSadella** [reply]
It was a Port Problem. It's working now with two sources at once and the HDR/SDR switch is triggered. 

You are a LEGEND.

**Reply by awawa-dev** [reply]
Thanks for the feedback. Nice to know that automation from webos works even when used on another device with USB grabber.
